### ISSUE TITLE
Virtual map reconnect: improve virtual subtree snapshots on the teacher side

### ISSUE THREAD

**Issue by artemananiev** [hashgraph/hedera-services#13732]
During reconnects, teachers make a snapshot for every virtual map in the state tree, before it starts sending it to learners. It starts with a call to `VirtualPipeline.detachCopy()` from `TeacherPullVirtualTreeView` or `TeacherPushVirtualTreeView` constructor and ends with `VirtualRootNode.detach()` This method does two things:

1. Creates a node cache snapshot
2. Creates a data source snapshot (using the current data source builder)

Node cache snapshot is more or less fine. It creates a new Java object and migrates only relevant mutations from the existing cache to it. Data source snapshot is something we can improve. Currently MerkleDb data source builder just calls `MerkleDbDataSource.snapshot()`, which hard-links all data files and dumps all indices to disk. Before such a snapshot can be used, all these indices must be loaded from disk, and it takes time.

@jasperpotts suggests that we can avoid index serialization/deserialization in this case.
